src/backend/distributed/commands/table.c

```diff
@@ -1253,10 +1253,13 @@ PreprocessAlterTableStmt(Node *node, const char *alterTableCommand,
 					deparseAT = true;
 					useInitialDDLCommandString = false;
 
-					/* the new column definition will have no constraint */
+					/*
+					 * Since we don't support constraints for AT_AddColumn
+					 * we have to set is_not_null to true explicitly for identity columns
```

Comment:
I can't quite follow the comment. What is in the `constraints` field that makes us want to set it to NULL?

Are there 2 different ways of expressing NOT NULL?

Reply:
The identity column has a constraint named CONSTR_IDENTITY (see line 1251) which makes the column not nullable. Since we don't support constraints for AT_AddColumn. I set is_not_null to true to make it not nullable.

Reply:
> Are there 2 different ways of expressing NOT NULL?

Yes. If we have the constraint `CONSTR_IDENTITY`, `columnDef->is_not_null` is actually false. Having `CONSTR_IDENTITY` is enough on its own to have the not null condition defined for the column, so PG doesn't deem it necessary to set the `is_not_null` field
So , the constraints field has a single value which is `CONSTR_IDENTITY`, and that's it. We don't want that in the deparser, so we set the constraints field to NULL. However, now we do need to set the `is_not_null` field, to attain this condition for the newly defined column.

Reply:
@gokhangulbiz you can modify the comment a little bit to explain that `is_not_null` is false when `CONSTR_IDENTITY` is found, so after we remove `CONSTR_IDENTITY`, we need to set `is_not_null`. You shouldn't to say that we don't support it in the deparser, because we don't want to deparse `CONSTR_IDENTITY` anyway - that's the point - we don't want to send it to the shards. So, our deparser is fine.